src/components/ToDoItem.vue

```diff
@@ -97,19 +98,27 @@ function toggleEditState() {
   isFormEditable.value = !isFormEditable.value
 }
 
-function checkToDo(checked: boolean, updatedToDo: Todo, index: number) {
-  props.todo.isChecked = checked
-  editToDo(updatedToDo, index)
+function checkToDo(checked: boolean) {
+  toggleEditState()
+  emit('toggleCheck', checked, props.todo.id)
 }
 
-function editToDo(todo: Todo, index: number) {
+function editToDo(todo: Todo) {
   toggleEditState()
-  emit('editToDo', todo, index)
+  if (props.todo.isChecked) {
```

Comment:
you can write this in one line `emit('editToDo', todo, props.todo.isChecked ? 'complete' : 'incomplete')`